### DISCUSSION TITLE
How to view which rows comprise a topic

### DISCUSSION THREAD

**Discussion by ecksteing** [MaartenGr/BERTopic#663] [Q&A] [answered]
Is it possible to view the rows that comprise a topic? If so how?
For example, say I have a dataframe with a column called "ID" that is unique for each row, and the text to model is in a column titled "TEXT". How do I view which rows are deemed to constitute each topic?
 I'm more familiar with PyCaret and .head shows the dataframe columns and assigned topic number. Any help appreciated :)

**Comment by MaartenGr** [comment, marked as answer]
The package follows, to a certain extent, sklearn's API in that whenever you use `transform` on a set of documents, it will return the topics in the same order. Let's say you have the following code:

```python
from bertopic import BERTopic
from sklearn.datasets import fetch_20newsgroups

docs = fetch_20newsgroups(subset='all',  remove=('headers', 'footers', 'quotes'))['data']

topic_model = BERTopic()
topics, probs = topic_model.fit_transform(docs)
```
Here, `docs` is a list of documents on which you train the model. Running `.fit_transform(docs)` will return the variable `topics`. In `topics`, you will find the topics that belong to each documents. The topic in `topics[0]` corresponds to the document in `docs[0]`, `topics[1]` to `docs[1]`, etc. 

You can use that structure to extract the documents under a certain topic by using, for example, the following:

```python
import pandas as pd
results = pd.DataFrame({"Doc": docs, "Topic": topics})
```

The `results` variable can then be extended with whatever metadata you have, like the `ID` of each document. 

If you already have a data frame, `df`, with TEXT and ID columns, then you would only need to do the following:

```python
df["Topic"] = topics
```

**Reply by ecksteing** [reply]
Thank you Maarten. Worked perfectly and very much appreciated!
